python/ray/autoscaler/updater.py

```diff
@@ -159,20 +160,23 @@ def run(self):
         self.exitcode = 0
 
     def wait_for_ssh(self, deadline):
-        logger.info("NodeUpdater: "
-                    "{}: Waiting for SSH...".format(self.node_id))
 
+        waiting_msg = "NodeUpdater: {}: Waiting for SSH.".format(self.node_id)
+        logger.info(waiting_msg)
+        retries = 0
         while time.time() < deadline and \
                 not self.provider.is_terminated(self.node_id):
             try:
-                logger.debug("NodeUpdater: "
-                             "{}: Waiting for SSH...".format(self.node_id))
-
                 # Setting redirect=False allows the user to see errors like
                 # unix_listener: path "/tmp/rkn_ray_ssh_sockets/..." too long
                 # for Unix domain socket.
-                self.ssh_cmd("uptime", connect_timeout=5, redirect=False)
-
+                logger.info(waiting_msg + "." * retries)
+                with open("/dev/null") as devnull:
```

Comment:
We cannot simply redirect things to /dev/null.

Reply:
Recommending [redict_stdout](https://docs.python.org/3.4/library/contextlib.html#contextlib.redirect_stdout)

```
f = io.StringIO()
with redirect_stdout(f):
    help(pow)
s = f.getvalue()
```